pkg/controller/acmechallenges/scheduler/scheduler.go

```diff
@@ -69,6 +69,9 @@ func (s *Scheduler) scheduleN(n int, allChallenges []*cmacme.Challenge) ([]*cmac
 
 	numberToSelect := n
 	remainingNumberAllowedChallenges := s.maxConcurrentChallenges - inProgressChallengeCount
+	if remainingNumberAllowedChallenges < 0 {
+		remainingNumberAllowedChallenges = 0
+	}
```

Comment:
I suggest a unit test for this. It looks like there's an existing unit test suite covering the public `ScheduleN` function. Perhaps we could add a new table entry to those tests with more challenges than the setting, or with `maxConcurrentChallenges = 0`.

Reply:
Added a test

Reply:
I can't see the test. Did you forget to push it?